imagersite/imager_images/views.py

```diff
@@ -1,68 +1,151 @@
+            'photos': photos,
+        })
 
+        return context
 
-def photo_view(request):
-    """Define the library view."""
-    public_photos = Photo.objects.filter(published='PUBLIC')
 
-    context = {
-        'public_photos': public_photos,
-    }
+class AlbumView(ListView):
+    """Define the album view class."""
 
-    return render(request, 'imager_images/photo.html', context)
+    template_name = 'imager_images/album.html'
+    model = Album
 
+    def get(self, *args, **kwargs):
+        """Check that user is authenticated, get args and kwargs."""
+        if not self.request.user.is_authenticated:
+            return redirect('home')
 
-def album_detail_view(request, id=None):
-    """Define the library view."""
-    this_album = Album.objects.filter(id=id).first()
+        return super().get(*args, **kwargs)
 
-    context = {
-        'this_album': this_album,
-    }
+    def get_context_data(self, **kwargs):
+        """Filter the context for display."""
+        context = super().get_context_data(**kwargs)
 
-    return render(request, 'imager_images/album_detail.html', context)
+        public_albums = Album.objects.filter(published='PUBLIC')
 
+        context = {
+            'public_albums': public_albums,
+        }
 
-def photo_detail_view(request, id=None):
-    """Define the library view."""
-    this_photo = Photo.objects.filter(id=id).first()
+        return context
 
-    context = {
-        'this_photo': this_photo,
-    }
 
-    return render(request, 'imager_images/photo_detail.html', context)
+class PhotoView(ListView):
+    """Define the album view class."""
+
+    template_name = 'imager_images/photo.html'
+    model = Photo
+
+    def get(self, *args, **kwargs):
+        """Check that user is authenticated, get args and kwargs."""
+        if not self.request.user.is_authenticated:
+            return redirect('home')
+
+        return super().get(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        """Filter the context for display."""
+        context = super().get_context_data(**kwargs)
+
+        public_photos = Photo.objects.filter(published='PUBLIC')
+
+        context = {
+            'public_photos': public_photos,
+        }
+
+        return context
+
+
+class AlbumDetailView(DetailView):
+    """Define the album detail view class."""
+
+    template_name = 'imager_images/album_detail.html'
+    model = Album
+    context_object_name = 'album'
+    slug_url_kwarg = 'id'
+    slug_field = 'id'
+
+    def get(self, *args, **kwargs):
+        """Check that user is authenticated, get args and kwargs."""
+        if not self.request.user.is_authenticated:
+            return redirect('home')
+
+        return super().get(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        """Filter the context for display."""
+        context = super().get_context_data(**kwargs)
+        id = context['album'].id
+        this_album = Album.objects.filter(id=id).first()
```

Comment:
You do not need to do this, because the `Album` specified by your identifier (the slug or pk kwarg) is automatically retrieved by the `DetailView` class.